test/e2e/generator/generate.go

```diff
@@ -107,22 +109,42 @@ type Options struct {
 func generateTestnet(r *rand.Rand, opt map[string]interface{}) (e2e.Manifest, error) {
 	manifest := e2e.Manifest{
 		IPv6:             ipv6.Choose(r).(bool),
-		ABCIProtocol:     nodeABCIProtocols.Choose(r),
 		InitialHeight:    int64(opt["initialHeight"].(int)),
 		InitialState:     opt["initialState"].(map[string]string),
 		Validators:       &map[string]int64{},
 		ValidatorUpdates: map[string]map[string]int64{},
 		Nodes:            map[string]*e2e.ManifestNode{},
 		KeyType:          keyType.Choose(r).(string),
 		Evidence:         evidence.Choose(r).(int),
-		QueueType:        opt["queueType"].(string),
+		QueueType:        "priority",
 		TxSize:           txSize.Choose(r).(int),
 	}
 
 	if voteExtensionEnabled.Choose(r).(bool) {
 		manifest.VoteExtensionsEnableHeight = manifest.InitialHeight + voteExtensionEnableHeightOffset.Choose(r).(int64)
 	}
 
+	if opt["abci"] == "builtin" {
+		manifest.ABCIProtocol = string(e2e.ProtocolBuiltin)
+	} else {
+		manifest.ABCIProtocol = ABCIProtocols.Choose(r)
+	}
+
+	switch abciDelays.Choose(r).(string) {
+	case "none":
+	case "small":
+		manifest.PrepareProposalDelayMS = 100
+		manifest.ProcessProposalDelayMS = 100
+		manifest.VoteExtensionDelayMS = 20
+		manifest.FinalizeBlockDelayMS = 200
+	case "large":
+		manifest.PrepareProposalDelayMS = 200
+		manifest.ProcessProposalDelayMS = 200
```

Comment:
In the "large" case, I would assume `PrepareProposal` and `ProcessProposal` are doing "immediate execution", so I'd set the delay to something similar to `FinalizeBlock`.
Just an idea, up to you.

Reply:
Aren't we explicitly not supporting immediate execution? We can still extend these out a little. They're so far quite arbitrary values

Reply:
By "immediate execution", I'm referring to the fact that they (`PrepareProposal`/`ProcessProposal`) fully execute the block, without waiting for `FinalizeBlock` to do it.
(N.B.: Even if they execute the block immediately, some of the the block's header fields, like `app_hash`, are still referring to the previous block, which is not supported in v0.36)